### ISSUE TITLE
Some invocations to a TLS enabled daemon occur via HTTP

### ISSUE THREAD

**Issue by kyleschlosser** [docker/swarm#241]
I have docker nodes and swarm daemon configured to use TLS, and some commands are working against the swarm daemon such as ps, but when I try to create a container, the container is created but does not start. It appears the start invocation is using http, while the other invocations which are succeeding presumably are using https. 

```
docker --tlsverify --tlscacert=... --tlskey=... --tlscert=... -H ... run -dP nginx
32ea1eb057129ab2d37f511469ba5033357efc338138ce676b5a1bc7f3a19863
FATA[0006] Error response from daemon: Post http://x.x.x.x:2376/v1.16/containers/32ea1eb057129ab2d37f511469ba5033357efc338138ce676b5a1bc7f3a19863/start: malformed HTTP response "\x15\x03\x01\x00\x02\x02\x16"
```

Swarm Log: 
```
INFO[1839] GET /v1.16/containers/json?all=1
INFO[2176] POST /v1.16/containers/create
INFO[2176] event -> status: "create" from: "nginx:1" id: "32ea1eb057129ab2d37f511469ba5033357efc338138ce676b5a1bc7f3a19863" node: "sp-dev-docker01"
INFO[2177] POST /v1.16/containers/32ea1eb057129ab2d37f511469ba5033357efc338138ce
676b5a1bc7f3a19863/start
ERRO[2177] Post http://x.x.x.x:2376/v1.16/containers/32ea1eb057129ab2d37f511
469ba5033357efc338138ce676b5a1bc7f3a19863/start: malformed HTTP response "\x15\x03\x01\x00\x02\x02\x16"
```

Version reported is 0.0.1 installed via go get -u github.com/docker/swarm earlier today.

**Comment by vieux** [comment]
@kyleschlosser thanks for the report, can you test the fix in #242 ?

**Comment by kyleschlosser** [comment]
Thanks for the quick turn around, its working fine now.
